### PR TITLE
Divyanshu | Stage | Implementation of Zimbabwe VAT in GIDDH - Version 2

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cv0mmu4

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.html [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    </div>
+    <div class="detailed-liability-report mx-auto pd-15">
+        <div class="mt-2">
+            <button mat-stroked-button (click)="backToOverViewReport()">
+                <i class="fa fa-angle-left font-16 mr-1"></i>
+                <span>{{ commonLocaleData?.app_back }}</span>
+            </button>
+        </div>
+        <div class="table-responsive mt-4">
+            <table mat-table [dataSource]="vatLiabilityDetailedReport.sections"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.html]:
```suggestion
            <table mat-table [dataSource]="vatLiabilityDetailedReport?.sections"
```

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        section: '',
+        currencyCode: 'BWP'
+    };
+    /** True if API is calling in progress */
+    public isLoading: boolean = false;
+    /** This will hold local JSON data */
+    public localeData: any = {};
+    /** This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** This will hold the value out/in to open/close setting sidebar popup */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts]: Correct the documentation

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     * This will be use for component initialization
+     *
+     * @memberof LiabilityDetailedReportComponent
+     */
+    public ngOnInit(): void {
+        document.querySelector('body').classList.add('gst-sidebar-open');
+
+        this.route.queryParams.pipe(takeUntil(this.destroyed$)).subscribe(params => {
+            if (params) {
+                this.vatLiabilityReportRequest.from = params['from'];
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts]:
```suggestion
                this.vatLiabilityReportRequest.from = params.from
                
                correct for this all;
```

File: apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                                <ng-template #clickableItem>
+                                    <a href="javascript:;" (click)="viewDetailedReport(vatReport)"> 
+                                        {{vatReport?.section }}
+                                    </a>
+                                </ng-template>
+                            </td>
+                            <td width="20%" class="text-right">
+                                <amount-field 
+                                    class="d-inline-block" 
+                                    [amount]="vatReport?.vattotal"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.html]: Change vattotal to camel case , also make changes in API

File: apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+})
+
+export class LiabilityReportComponent implements OnInit, OnDestroy {
+    /** Observable to unsubscribe all the store listeners to avoid memory leaks */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** This will hold local JSON data */
+    public localeData: any = {};
+    /** This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** Holds Fromgroup  */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts]: Correct this documentation

File: apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    public isTaxApiInProgress: boolean;
+    /** Holds Table Data  */
+    public vatLiabilityOverviewReport: any[];
+    /** Holds Active company details */
+    public activeCompany: any = null;
+    /** Stores the current branch */
+    public currentBranch: any = { name: '', uniqueName: '' };
+
+    constructor(
+        private formBuilder: UntypedFormBuilder,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts]: Make it FormBuilder in over all page

File: apps/web-giddh/src/app/vat-report/transactions/vat-report-transactions.component.html [hunk 7]
```diff
@@ -31,6 +32,7 @@
                     mat-table
                     [dataSource]="vatReportTransactions.results"
                     class="table basic table-bordered mt-2 invoice-table desktop-table"
+                    *ngIf="vatReportTransactions.results?.length && !isLoading"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/transactions/vat-report-transactions.component.html]:
```suggestion
                    *ngIf="vatReportTransactions?.results?.length && !isLoading"
```

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts [hunk 8]
```diff
@@ -0,0 +1,592 @@
...
+    }
+
+    /**
+     * Save last currency in which the report was viewed.
+     *
+     * @memberof VatReportFiltersComponent
+     */
+    public saveSelectedCurrency(currencyCode: string): void {
+        let request = {
+            module: this.moduleType,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]:
```suggestion
            module: this.moduleType
```

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts [hunk 9]
```diff
@@ -0,0 +1,592 @@
...
+        const isFirstDayOfMonth = from.isSame(fromStartOfMonth, 'day');
+        const isLastDayOfMonth = to.isSame(toEndOfMonth, 'day');
+        const isSameMonthAndYear = from.isSame(to, 'month') && from.isSame(to, 'year');
+
+        if (isFirstDayOfMonth && isLastDayOfMonth && isSameMonthAndYear) {
+            return {
+                status: true,
+                monthName: from.format('MMMM'),
+                monthNumber: from.month() + 1,
+                year: from.year(),
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]:
```suggestion
                year: from.year()
```

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts [hunk 10]
```diff
@@ -0,0 +1,592 @@
...
+            };
+        } else {
+            return {
+                status: false,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]:
```suggestion
                status: false
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.html (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/liability-detailed-report/liability-detailed-report.component.ts]: Done for all

File: apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.html (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/liability-report/liability-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/transactions/vat-report-transactions.component.html (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/transactions/vat-report-transactions.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 9 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 10 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**